code/socialDistribution/models/post.py

```diff
@@ -85,6 +85,12 @@ class Visibility(models.TextChoices):
     #     blank=True,
     # )
 
+    image = models.ForeignKey('Image',
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE
+    )
+
```

Comment:
Can't we save the image as a base64 encoding in the content BinaryField? I only say this because InboxPosts (posts that are received by the user) don't have this field.

Reply:
I think Michelle told me about that at some point but to be honest, this completely escaped my mind. 😅 I'll make the changes

I'm also wondering about something related to that. Since we allow the user to upload image files on top of having content text, should we create a separate post for holding that image and then embed the url to that image post at the end of the content (text) of the first post. If we do that, we can get rid of references to images inside the Post models and store every content as binary.

When the post is rendered, the js code handling markdown will make a GET request for the uploaded image post as for regular markdown.

Does that make sense?

Reply:
> I'm also wondering about something related to that. Since we allow the user to upload image files on top of having content text, should we create a separate post for holding that image and then embed the url to that image post at the end of the content (text) of the first post. If we do that, we can get rid of references to images inside the Post models and store every content as binary.

Yes, exactly. I think storing everything as binary is the way to go. OR ( because it's simpler), we just force the user to make either a text post OR an image post, but not both. That way the user has to make two posts on their own and manually embed whatever image they want to add. I think this would be quicker and easier.



> When the post is rendered, the js code handling markdown will make a GET request for the uploaded image post as for regular markdown.

Not sure if I totally understand, but yes I think so. The markdown will reference the image URL that is hosted by our server, or any other image URL that the user adds to their markdown post. I think it would be best if the image posts were hosted from a URL different from /author/id/post/id. The /post/id endpoint should probably only be for the JSON post objects with the base64 encoded content field.

e.g. you could make an /media/ path and serve the images as images from there (if you don't do that already, I'm just not totally sure if I understood.) GETing from this /media/{id} path could render the binary content of post with id={id} as an image, **only if that post has content_type == "image or whatever".**

At least those are my thoughts, you're free to see how to best get this working. Let me know if that doesn't make sense.

Reply:
Yup, you're right. I think I'll go with the second approach since it seems easier and cleaner (I think). That is, have allow the user to create only one type of post (either image or text/md).